packages/main/src/plugin/dockerode/libpod-dockerode.spec.ts

```diff
@@ -211,13 +213,18 @@ test('Check attach API', async () => {
   server.listen({ onUnhandledRequest: 'error' });
 
   const api = new Dockerode({ protocol: 'http', host: 'localhost' });
-  const libPod = api as any;
+  const libPod = api as unknown as DockerodeInternals;
 
   // patch libpod to not wait for the test as websocket is not supported by mock server
   const originalBuildRequest = libPod.modem.buildRequest;
-  libPod.modem.buildRequest = function (options: unknown, context: any, data: unknown, callback: unknown): void {
+  libPod.modem.buildRequest = function (
+    options: RequestOptions,
+    context: DockerModem.DialOptions,
+    data?: string | Buffer | NodeJS.ReadableStream,
+    callback?: DockerModem.RequestCallback,
+  ): void {
     context.openStdin = false;
-    return originalBuildRequest.call(this, options, context, data, callback);
```

Comment:
Are we sure the build request do not need to be awaited ? here if call is a promise, it become a dangling promise, I see the return type of `buildRequest` being void, but is it really?

Reply:
yes it's the old school pattern (callback pattern)

Reply:
omg, sorry I did see the callback